### PR TITLE
added the hover provider

### LINKED ISSUE #343: `About` on hover
Can work only partially with today's Coq API, so we should disable it by default.

### PR DESCRIPTION
**Pull request by rtetley**
Do an 'About' query on hover. 
This implementation works, however it made me question the sanity of the responses sent by the server ? @maximedenes it seems that the server really send back a simple string with result instead of an object containing an id and a result ? Maybe it is my misunderstanding of how the protocol works ?

Closes #343.

### CODE REVIEW

**Comment by Hjaltesorgenfrei** [comment]
It is my understanding that hover should be provided by the backend so that it can be used by other editors which use the LSP. So the backend should implement `textDocument/hover`, <https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_hover>

**Comment by rtetley** [comment]
Both implementations are valid options: 
https://code.visualstudio.com/api/language-extensions/programmatic-language-features
However I agree with you that the standard lsp option allows to support more editors. 
In this iteration I chose this option as I wasn't sure we wanted to do an about query by default on hover. 
After discussion with @maximedenes it seems to be the route we will be taking !

**Comment by rtetley** [comment]
This also calls for adding client Capabilities in the initialize sequence !

**Comment by rtetley** [comment]
As suggested by @Hjaltesorgenfrei re-did the implementation, we now handle `textDocument/hover` in the backend.
Will probably need some tweeks.

File: language-server/lsp/lspData.mli [hunk 1]
```diff
@@ -57,4 +57,22 @@ module ServerCapabilities : sig
     hoverProvider : bool;
   } [@@deriving yojson]
 
-end
\ No newline at end of file
+end
+
+module HoverResponse : sig
```
[the remark this answers is not in the document]
Comment (rtetley) [reply language-server/lsp/lspData.mli:62]: Not clear to me. It seems they might have just been using this as an example of how the protocol works ? If that's the case I really don't know why they didn't use the real response. Anyways, I'll remove it.

**Review by maximedenes — APPROVED** [review]

**Pull request merged**